public/script.js

```diff
@@ -1613,47 +1607,42 @@ async function Generate(type, automatic_trigger, force_name2) {
                 //if (is_pygmalion && i == chat2.length-1) item='<START>\n'+item;
                 arrMes[arrMes.length] = item;
             } else {
-                console.log('reducing chat.length by 1');
-                i = chat2.length - 1;
+                break;
             }
 
             await delay(1); //For disable slow down (encode gpt-2 need fix)
-            // console.log(i+' '+chat.length);
-
-            count_exm_add = 0;
-
-            if (i === chat2.length - 1) {
```

Comment:
This is a check for the last iteration of the loop. Considering that `chat2` is guaranteed to have at least 1 item (`chat2.push('');` in line 1601) and after this block there is nothing besides `i++;`, it can simply be moved out of the loop and out of the `if`.